### PR TITLE
Feature/its the item circle of life

### PR DESCRIPTION
**Pull request by SilliusSodus**
The nourishment mug now changes to a foamy version when it is 100% full and switches back to an empty version when it goes below 0%. Check out the mug + foam prefab to see how it works.

### CODE REVIEW

File: behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs [hunk 1]
```diff
@@ -40,33 +41,33 @@ private void OnTriggerEnter(Collider col)
...
 
     private void PrepareDish()
     {
         string Dish = this.gameObject.name;
         isPreparing = true;
-        Debug.Log("The " + Dish + " is " + _fullness + "0% full.");
+        changeFullness(1);
         _doneness += 1;
-        _fullness += 1;
+        Debug.Log("The " + Dish + " is " + _fullness/stats.capacity + " full.");
```
Comment (TJ-Smits) [inline behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs]: Not every word has to be split up into one + " " + but this is a really minor thing

Comment (MattijnP) [reply behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs]: You can use string interpolation like this:
```cs
Debug.Log($"The {Dish} is {_fullness / stats.capacity} full.");
```

File: behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs [hunk 2]
```diff
@@ -107,10 +109,30 @@ private void HappyWithOrder(NPC npc)
 
     private void consume(NPC npc)
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs]: Method names should start with a capital letter

File: behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs [hunk 3]
```diff
@@ -107,10 +109,30 @@ private void HappyWithOrder(NPC npc)
...
     {
-        _fullness--;
+        changeFullness(-1);
         Debug.Log("*glug glug* My consumption is " + _fullness + "0% full");
     }
 
+    private void changeFullness(float change)
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs]: Method names should start with a capital letter

Comment (TJ-Smits) [reply behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs]: All methods should start with a Capital letter. Private or not

File: behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs (hunk 2 above)
Comment (TJ-Smits) [reply behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs]: All methods should start with a Capital letter. Private or not

File: behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs (hunk 1 above)
Comment (SilliusSodus) [reply behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs]: Done

File: behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs (hunk 2 above)
Comment (SilliusSodus) [reply behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs]: Fixed

File: behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs (hunk 3 above)
Comment (SilliusSodus) [reply behind-the-tankard/Assets/Scripts/Interactable/Categories/Nourishment.cs]: Same, fixed.

**Pull request merged**